### DISCUSSION TITLE
【依頼】モータ制御について

### DISCUSSION THREAD

**Discussion by tsukada-shigeki** [toppers/users-forum#78] [TOPPERS Seminar]
竹内様
お世話になります。
セミナーの範囲外で申し訳ありません。
toppersを用いてモータ制御を考えています。
１．依頼内容を添付ワードファイルにまとめました。
２．依頼資料をまとめて添付しました。
　　１）adc1を改造して2チャンネルリードできるものを
　　　　adc4orgフォルダに入れました。この変更点を、パワポでまとめました。
　　２）adc4org+PWMをトライしています（adc4フォルダ参照ください）。
　　３）エラーの解決方法をご教示願います。
お時間のある時で結構ですのでアドバイスをお願い致します。
以上宜しくお願い致します。
塚田
[モータ制御依頼.docx](https://github.com/toppers/users-forum/files/9032181/default.docx)
[モータ制御資料.zip](https://github.com/toppers/users-forum/files/9032182/default.zip)

**Comment by Ryohsuke-Takeuchi** [comment]
adc4については、adctest.cとMakefileを以下の修正でビルドエラーはでなくなります。正しく、実行できるかは、わかりませんが。
![image](https://user-images.githubusercontent.com/91311953/177071339-9ce78bba-cac9-4027-89d9-2b0ed1d19f61.png)

**Reply by tsukada-shigeki** [reply]
竹内様
お世話になります。
早速のアドバイス有難うございます。

makefile の変更点は理解しました。
adtest,cで#if 0 と　#endifを挿入すべき行がわかりません。
誠に申し訳ありませんが修正後のadtest.cのみ添付お願い致します。
以上宜しくお願い致します。
塚田

**Reply by Ryohsuke-Takeuchi** [reply]
以下の通りです
[adctest.zip](https://github.com/toppers/users-forum/files/9036572/adctest.zip)

**Reply by tsukada-shigeki** [reply]
竹内様
お世話になります。
ファイル拝受致しました。有難うございます。
只今より試します。
塚田

**Reply by tsukada-shigeki** [reply]
竹内様
お世話になります。

頂いたadctest.cをビルドし実行致しました。
お蔭様で、adc２入力と、pwm出力できました。
あとは速度制御を試します。有難うございました。
塚田
